src/api/app/models/user_session.rb

```diff
@@ -0,0 +1,29 @@
+class UserSession
+  class FailedAuthentication < StandardError; end
+  class DisabledAccount < StandardError; end
+
+  # Returns the user storaged in current Thread
+  #
+  # @return [User|nil] user for current session or nil if none
+  def self.current
+    Thread.current[:user]
+  end
+
+  # Storages the user in current Thread
+  #
+  # @param [User|nil]
```

Comment:
well, I can also give whatever I want to this method. I don't think if it has sense to specify what you expect. Also the variable speaks by itself, so again it can be removed it.